src/integrations.py

```diff
@@ -32,3 +42,39 @@ def load(cls, provider: AuthProxyProvider) -> "AuthProxyIntegrationData":
             authenticated_emails=authenticated_emails,
             authenticated_email_domains=authenticated_email_domains,
         )
+
+
+class TrustedCertificatesTransferIntegration:
+    def __init__(self, charm: CharmBase):
+        self._charm = charm
+        self._container = charm._container
+        self._peers = charm._peers
+        self.cert_transfer_requires = CertificateTransferRequires(charm, relationship_name="receive-ca-cert")
+
+    def update_trusted_ca_certs(self) -> None:
+        """Receive trusted certificates from the certificate_transfer integration.
+
+        This function is needed because relation events are not emitted on upgrade,
+        and because there is no persistent storage for certs.
+        """
+        if not self._charm.model.get_relation(relation_name=self.cert_transfer_requires.relationship_name):
+            logger.warning(
+                "Missing certificate_transfer integration, OAuth2 Proxy will skip validation of certificates presented when using HTTPS providers. Don't do this in production"
+            )
+            return
+
+        logger.info(
+            "Pulling trusted ca certificates from %s relation.",
+            self.cert_transfer_requires.relationship_name,
+        )
+        certs = []
+        for relation in self._charm.model.relations.get(self.cert_transfer_requires.relationship_name, []):
+            for unit in set(relation.units).difference([self._charm.app, self._charm.unit]):
+                cert_path = TRUSTED_CA_TEMPLATE.substitute(rel_id=relation.id)
+                if cert := relation.data[unit].get("ca"):
+                    self._container.push(cert_path, cert, make_dirs=True)
+                    certs.append(cert_path)
+
+        self._peers.data[self._charm.app][CERT_PATHS_KEY] = str(certs)
+
+        subprocess.run(["update-ca-certificates", "--fresh"])
```

Comment:
To add on that, I think we need to either update the default Go trust sources (`/usr/share/ca-certificates`) OR use the `provider_ca_files` config

Reply:
I chose to use `provider_ca_files`, do you think I should get rid of it?